### PR TITLE
feat: replace *jsdoc* plugin by *tsdoc* pluglin

### PR DESCRIPTION
**Pull request by rg911**
feat: replace *jsdoc* plugin by *tsdoc* pluglin

### CODE REVIEW

File: src/core/Address.ts [hunk 1]
```diff
@@ -28,15 +28,13 @@ export abstract class Address {
...
-     *
-     * @returns {Uint8Array}
      */
     public abstract getAddressBytes(): Uint8Array;
 
     /**
      * Get address in the encoded format ex: NAR3W7B4BCOZSZMFIZRYB3N5YGOUSWIYJCJ6HDFH.
      *
-     * @returns {string} Encoded addres string
+     * @returns Encoded addres string
```
Comment (Jaguar0625) [inline src/core/Address.ts:37]: is there a way to add spell checker :D 
and enforce periods (or not)?

Comment (rg911) [reply src/core/Address.ts:37]: 😄 , will have a look

File: src/core/symbol/VotingKeysGenerator.ts [hunk 2]
```diff
@@ -22,17 +22,17 @@ export class VotingKeysGenerator {
...
      */
     constructor(private readonly rootKeyPair: KeyPair, private readonly keyPairGenerator = SymbolKeyPair.generate) {}
 
     /**
      * Generate Symbol voting keys
      *
-     * @param {number} startEpoch Start voting key epoch
-     * @param {number} endEpoch End voting key epoch
-     * @returns {Uint8Array} Voting keys buffer
+     * @param startEpoch - Start voting key epoch
```
Comment (fboucquez) [inline src/core/symbol/VotingKeysGenerator.ts:33]: Do we need the dashes?

File: src/core/Address.ts (hunk 1 above)
Comment (rg911) [reply src/core/Address.ts:37]: the spell checker plugin we had before was not smart enough, it flags lots reserved words

Comment (segfaultxavi) [reply src/core/Address.ts]: I use CSpell and Language Tool with VSCode, and add keywords to my personal dictionary. After a while they know all of them :)

Comment (fboucquez) [reply src/core/Address.ts]: I agree with @Jaguar0625 

I think like the style checker and style formatter, having a ready-to-run cli spellcheck is super nicer. If not everybody would have their own dictionary and tools that may not be in sync (different IDEs, different reserved words) and cannot be validated in PRs.

We should somehow have a shared dictionary and add the spellchecker to each build tool. 
One example https://www.npmjs.com/package/eslint-plugin-spellcheck.

There are going to be a lot of typos, but well, need to fix them

Comment (fboucquez) [reply src/core/Address.ts]: @segfaultxavi can you share your dictionary? So we all use the same words. We probably need to push that somewhere in github and each language should use the right tool for the spellchecking

Comment (segfaultxavi) [reply src/core/Address.ts]: Sure... https://hackmd.io/lrT8XcxyRcapPCb7z_idEQ?view

File: src/core/symbol/VotingKeysGenerator.ts (hunk 2 above)
Comment (rg911) [reply src/core/symbol/VotingKeysGenerator.ts:33]: yeah, needed for the tsdoc plugin

File: src/core/Address.ts (hunk 1 above)
Comment (fboucquez) [reply src/core/Address.ts]: @Jaguar0625 I have craeted https://github.com/symbol/sdk-typescript/issues/37. Does it make sense to fix this in another PR?

**Review by fboucquez — APPROVED** [review]

**Pull request merged**